### PR TITLE
clarify select! doc

### PR DESCRIPTION
**Pull request by simonbyrne**

### CODE REVIEW

File: base/docs/helpdb/Base.jl [hunk 1]
```diff
@@ -1168,8 +1168,9 @@ Partially sort the vector `v` in place, according to the order specified by `by`
 `rev` so that the value at index `k` (or range of adjacent values if `k` is a range) occurs
 at the position where it would appear if the array were fully sorted via a non-stable
 algorithm. If `k` is a single index, that value is returned; if `k` is a range, an array of
-values at those indices is returned. Note that `select!` does not fully sort the input
-array.
+values at those indices is returned.
+
+Note that `select!` does not fully sort the input array, but it does ensure that all are correctly ordered with respect to the value at index `k`.
```
Comment (kshyatt) [inline base/docs/helpdb/Base.jl]: Put this in a `!!! note` block, like in `statistics.jl`?

Also, maybe add a doctest example for the note?

Comment (simonbyrne) [inline base/docs/helpdb/Base.jl]: updated.

**Comment by simonbyrne** [comment]
Is that a bit clearer now?

**Comment by kshyatt** [comment]
I think so, but a doctest would be really welcome here.

**Comment by kshyatt** [comment]
Going to close since helpdb is gone and we switched to markdown. If the changes to the wording are still desired feel free to reopen w/o conflicts/make a new pr!

**Pull request closed without merging**